File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        str_x = str(x)
        if str_x[-1] == 0:
            return False
        left, right = 0, len(str_x)-1 
        while left < right:
            if str_x[left] != str_x[right]:
                return False
            left += 1
            right -= 1
        return True

File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x == 0) return true;
        if (x < 0 || x % 10 == 0) return false;
        int reverse = 0;
        while (x > reverse) {
            reverse *= 10;
            reverse += x % 10;
            x /= 10;
        }
        return x == reverse || x == reverse / 10;
    }
};
//class Solution {
//public:
//    bool isPalindrome(int x) {
//        if (x == 0) return true;
//        if (x < 0 || x%10 == 0) return false;
//        int reverse = 0;
//        while (x > reverse) {
//            reverse *= 10;
//            reverse += x % 10;
//            x /= 10;
//        }
//        return x == reverse || x == reverse / 10; 
//    }
//};